src/component/category-form/category-form.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import autoBind from '../../utils/autobind';
+
+const defaultState = {
+  name: '',
+};
+
+class CategoryForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = this.props.category || defaultState;
+    autoBind.call(this, CategoryForm);
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    this.props.onComplete(this.state);
+    this.setState(defaultState);
+  }
+
+  render() {
+    const buttonText = this.props.category ? 'Update' : 'Create';
+    return (
+      <div>
+        <h1>Budget Tracking Application</h1>
```

Comment:
You might want to move this to an outer component such as Dashboard so it doesn't get repeated when making a new component.